### PR TITLE
Make `bool`'s deserialization non-malleable

### PR DESCRIPTION
**Pull request by ValarDragon**
## Description

Makes the deserialization of bool non-malleable. Now the only strings that can be deserialized as a boolean are the encodings of `0u8, 1u8`.

closes: #95 
---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (main)
- [x] Linked to Github issue with discussion and accepted design OR have an explanation in the PR that describes this work.
- [x] Wrote unit tests
- [x] Updated relevant documentation in the code
- [x] Added a relevant changelog entry to the `Pending` section in `CHANGELOG.md`
- [x] Re-reviewed `Files changed` in the Github PR explorer

### CODE REVIEW

**Comment by Pratyush** [comment]
Small comment: instead of arbitrarily mutating the data, why not try mutating it to every possible byte value? That’ll remove the dependence on ark-ff

**Comment by ValarDragon** [comment]
That works for testing bool, but wouldn't generalize if we wanted to test this for any data type with a larger serialization.

We have no other data type implemented in serialize (AFAIK) which has a non-1-to-1 encoding to test for atm, so maybe that point is moot. (Also for the current API to generalize well, we'd probably need to change `valid_mutation` to return type `T` instead of a `valid/invalid` boolean)

Maybe it'd make sense to refactor this method out of tests, and do the above change, so other types (e.g. Fields) can check this?

File: serialize/Cargo.toml [hunk 1]
```diff
@@ -16,6 +16,10 @@ edition = "2018"
 ark-serialize-derive = { path = "../serialize-derive", optional = true }
 ark-std = { git = "https://github.com/arkworks-rs/utils", default-features = false }
 
+[dev-dependencies]
+ark-ff = { path = "../ff" }
```
Comment (Pratyush) [inline serialize/Cargo.toml:20]: I think you can just use `thread_rng()` from `rand`; no need to import `ark-ff` for this.

Comment (ValarDragon) [reply serialize/Cargo.toml:20]: thread_rng requires std https://docs.rs/crate/rand/0.7.0/source/src/lib.rs

Comment (Pratyush) [reply serialize/Cargo.toml:20]: How about `StdRng`? Given that we want to clean up this infrastructure in the future, I don't think we should add this dependency here (even if it results in some duplication rn)

Comment (ValarDragon) [reply serialize/Cargo.toml]: Done!

**Review by Pratyush — APPROVED** [review]

**Pull request merged**